src/main/java/com/fexl/circumnavigate/mixin/chunkHandle/MinecraftServerMixin.java

```diff
@@ -0,0 +1,75 @@
+package com.fexl.circumnavigate.mixin.chunkHandle;
+
+import com.fexl.circumnavigate.Circumnavigate;
+import com.fexl.circumnavigate.core.WorldTransformer;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.level.ChunkHolder;
+import net.minecraft.server.level.DistanceManager;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.level.ChunkPos;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+@Mixin(MinecraftServer.class)
+public abstract class MinecraftServerMixin {
+
+    @Shadow
+    public abstract Iterable<ServerLevel> getAllLevels();
+
+    @Unique private int tickCount = 0;
+
+    @Inject(method = "stopServer", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/MinecraftServer;waitUntilNextTick()V"))
+    public void stopServer(CallbackInfo ci) {
+        // timeout, after that just force it
+        if (tickCount++ > 1500) {
+            Circumnavigate.LOGGER.warn("Force stop");
+            for (ServerLevel level : getAllLevels()) {
+                int overBounds = 0;
+                int inBounds = 0;
+
+                DistanceManager distanceManager = level.getChunkSource().chunkMap.getDistanceManager();
+                var updatingChunkMap = new ArrayList<>(level.getChunkSource().chunkMap.updatingChunkMap.keySet());
+                Map<Integer, Integer> loadingLevelMap = new HashMap<>();
+
+                for (Long pos : updatingChunkMap) {
+                    ChunkPos chunkPos = new ChunkPos(pos);
+                    WorldTransformer transformer = level.getTransformer();
+                    if (transformer.isChunkOverBounds(chunkPos)) {
+                        overBounds++;
+                    } else {
+                        inBounds++;
+                    }
+
+                    ChunkHolder chunkHolder = distanceManager.getChunk(pos);
+                    if (chunkHolder != null) {
+                        int loadingLevel = chunkHolder.getTicketLevel();
+                        loadingLevelMap.put(loadingLevel, loadingLevelMap.getOrDefault(loadingLevel, 0) + 1);
+                    } else {
+                        Circumnavigate.LOGGER.error("ChunkHolder is null for chunk: {}", chunkPos);
+                    }
+
+                    level.getChunkSource().chunkMap.toDrop.add(pos);
```

Comment:
This is what vanilla code should do, but for some reason chunk loading level is too low. So this line adds all of left chunks to `toDrop` list which then vanilla code computes and actually unloads chunks.